src/firebolt/async_db/cursor.py

```diff
@@ -584,6 +591,49 @@ def setinputsizes(self, sizes: List[int]) -> None:
     def setoutputsize(self, size: int, column: Optional[int] = None) -> None:
         """Set a column buffer size for fetches of large columns (does nothing)."""
 
+    @check_not_closed
+    async def cancel(self, query_id: str) -> None:
+        """Cancel a server-side async query."""
+        try:
+            resp = await self._api_request(
+                parameters={"query_id": query_id}, url_suffix="cancel"
+            )
+            if resp.status_code == codes.BAD_REQUEST:
+                raise OperationalError(
+                    f"Asynchronous query {query_id} was not cancelled."
+                )
+        except Exception:
+            self._state = CursorState.ERROR
+            raise
+
+    @check_not_closed
+    async def get_status(self, query_id: str) -> QueryStatus:
+        """Get status of a server-side async query. Return the state of the query."""
+        try:
+            resp = await self._api_request(
+                parameters={"query_id": query_id}, url_suffix="status"
+            )
+            # if resp.status_code == codes.BAD_REQUEST:
+            #     raise OperationalError(
+            #         f"Asynchronous query {query_id} status check failed."
+            #     )
+            if resp.headers.get("content-length", "") == "0":
+                raise OperationalError("No response to asynchronous query.")
+            if resp.status_code == codes.BAD_REQUEST:
+                return QueryStatus.NOT_AVAILABLE
+            resp_json = resp.json()
+            if "status" not in resp_json:
+                raise OperationalError(
+                    f"Invalid response to asynchronous query: missing query ID."
```

Comment:
It probably should be "missing status" or something

Reply:
Done.